src/sql/connection.py

```diff
@@ -195,11 +195,19 @@ def _error_module_not_found(cls, e):
         return ModuleNotFoundError("test")
 
     def __init__(self, engine, alias=None):
-        self.dialect = engine.url.get_dialect()
-        self.metadata = sqlalchemy.MetaData(bind=engine)
+        self.url = engine.url
```

Comment:
So I looked into changing it from class variables to just defining them as `metadata=` and `dialect=` and it seems like it would actually be more convoluted doing it like so, because you'd have to save the engine as a part of the class and then check the conditional whenever you're accessing it.